File: designpatterns/src/com/renho/behavior/memento/Originator.java
package com.renho.behavior.memento;

public class Originator {

	private String state;
	
	public Memento createMemento() {
		return new Memento(state);
	}

	public String getState() {
		return state;
	}

	public void setState(String state) {
		this.state = state;
	}
	
	public void setMemento(Memento memento) {
		state = memento.getState();
	}
	
	public void show() {
		System.out.println("Originator:" + state);
	}
}

File: designpatternsCpp/src/behavior/memento/Originator.cpp
/*
 * Originator.cpp
 *
 *  Created on: 2015Äê4ÔÂ7ÈÕ
 *      Author: renho
 */
#include <iostream>
#include "Originator.h"
using namespace std;

Originator::Originator() {
	state = -1;
}

Originator::~Originator() {
	cout << "renho" << endl;
}

void Originator::setState(int s) {
	state = s;
}

int Originator::getState() {
	return state;
}

Memento* Originator::createMemento() {
	return new Memento(state);
}

void Originator::setMemento(Memento* m) {
	state = m->getState();
}

void Originator::show() {
	cout << state << endl;
}
